### ISSUE TITLE
To Do

### ISSUE THREAD

**Issue by sherlocknow** [sherlocknow/DSI16_P4_G4_#8]
====================================
Data Cleaning 
====================================
	- Merge sample and weather (Peter - Done)
	- Merge NumMosquitos > 50 rows back into one row (train.csv) (Peter - Done) 
	- understand Address accuracy (train.csv) (Dominic - Done)
	- check isnull. (Dominic - Done)
	- create data dictionary , (weather.csv) & (train.csv) (Cheyanne - Done)
	- splitting up categorical,ordinal,continuous, mapping the ordinal (Dominic - Done)
	- writing data dictionary (Cheyanne - Done)
	

====================================
Data Visualisation / EDA 
====================================

SNS plot on weather variables (weather.csv) (Vikas - Done)

**Barplot**
Types of mosquitoes w.r.t to WNV use Barplot (Vikas - Done)
- train.csv

**Locations use Geoplot**
- spray path (spray.csv), (Cheyanne - Done)
- station towers (data decsription) (Cheyanne - Done)
- traps location  ( train.csv) (Cheyanne - Done)

- relationship of weather parameters like temperature or other weather factors -> will result in growth in mosquitos ? (Vikas - Done)
- Especially with rain against mosquito, as rain is a breeding ground (Vikas - Done)

- are the species of mosquito clustered in a certain location ? (Vikas - Done)
	○ train.csv - species groupby number of mosquitoes
	○ train.csv - traps
		

	- Which strain of mosquito carry the virus ? (Vikas - Done)
	- Count of mosquito before and after spray date , to prove effectiveness of spray ? (Vikas - Done)
	- Do sea level affect the count of mosquito ? (Vikas - Done)
	Station 1: CHICAGO O'HARE INTERNATIONAL AIRPORT Lat: 41.995 Lon: -87.933 Elev: 662 ft. above sea level
	Station 2: CHICAGO MIDWAY INTL ARPT Lat: 41.786 Lon: -87.752 Elev: 612 ft. above sea level

====================================
Modelling 
====================================

Feature Selection
	- Check for coollinearity, if have remove the features (Dominic - Done)
	- Feature Engineering (Dominic - Done)

Feature Finalization 

	- PCA (Dominic - Done)
	- LR + Lasso Regularization to drop coefficient (Dominic - Done)
	- Get dummies one hot-encoding (Peter - Done)
	- Random forrest->feature importances (Cheyanne - Done)
	

===========
Modelling
===========
Points to note
	- Imbalanced class (Cheyanne - Done)
		○ SMOTE
	

Types of models 

	- XGBoost Classifier (Dominic - Done)
	XGBoost (eXtreme Gradient Boosting) is a decision-tree-based ensemble technique that uses a gradient boosting algorithm. Some key benefits of XGBoost includes the ability to regularize to reduce overfitting and its fast computational speed. The 'scale_pos_weight' hyperparameter helps to handle our imbalanced class. The recommended weight to use is the ratio of the number of negative class to the positive class.

	- Gradient Boosting Classifier (Dominic - Done)
	While the AdaBoost model identifies the shortcomings by using high weight data points, gradient boosting performs the same by using gradients in the loss function.
	
	- AdaBoost Classifer (Dropped this model)
	AdaBoost is a boosting ensemble method that fits a sequence of weak learners on modified versions of the data. AdaBoost works by putting more weight on misclassifications and less on those already handled well. This allows it to convert a set of weak classifiers into a strong one.
	
	- Bagging Classifier (Dropped this model)
	As decision trees tend to overfit on the training data, we will use bagged decision trees instead. Bagging is an ensemble method that conducts random resampling with replacement to create multiple models and aggregates the predictions.

	- Logistic Regression
	Logistic regression is one of the most commonly used methods for binary classification problems. We will apply logistic regression here to predict the probability of WNV presence

	- Random Forest (Cheyanne - Done)

	- KNN Classifier (Dropped this model)
	

====================================
Evaluation Points To Be Discussed 
====================================

- Stations 1 and 2 , both give weather data of Chicago, however one station is located at elevation from sealevel, so this might compromise our data)
